apps/server/src/services/integration-service/OscIntegration.ts

```diff
@@ -111,12 +112,9 @@ export class OscIntegration implements IIntegration<OscSubscription, OSCSettings
   }
 
   private stringToOSCArgs(argsString: string): ArgumentType[] {
-    // NOTE: regex taken from https://stackoverflow.com/questions/4031900/split-a-string-by-whitespace-keeping-quoted-segments-allowing-escaped-quotes
-    const parseRegex = /[\w.]+|"(?:\\"|[^"])+"/g;
+    const matches = splitWhitespace(argsString);
 
-    const matches = argsString.match(parseRegex);
-
-    const parsedArguments = [];
+    const parsedArguments = new Array<ArgumentType>();
 
```

Comment:
@alex-Arc if you want to redo the individual arg type/value parsing as well go ahead. The splitWhitespace doesn't return things in the same way as my previous method did so there will need to be some changes here (probably just remove the quoted string stuff) once you are done with the splitWhitespace stuff.

Reply:
i'm was looking around to try and get a way for us to force floats for `0.0` and the like
but found that our OSC lib will do a check to see if `0.0 == 0` and then convert it to an integer
this all happens in the Message contractor and since we can't get rid of that for now I think we should just let it do most of the work

Reply:
I did see that but I don't like that logic if the user types `1.0` it should be sent as a float NOT an integer. What am I doing right now bypasses all of that number parsing logic in `node-osc`.

Reply:
Also that link is to the old `node-osc` repo the relevant part is [here](https://github.com/MylesBorins/node-osc/blob/e6ab7013fd7de599a07c60920aab0749aff0a837/lib/Message.mjs#L25) in the new repo. If we pass in the complete arg object `{ type: 'f', value: 'something'}` then the library doesn't try to guess what argument type it should use.

Reply:
ahh sorry you are way ahead of me
i totally agree, think my OSC monitor cut of the floats so I thought it was not working

Reply:
Yeah, I did do some research before doing it the way I did it. I saw how the library parsed raw values into OSC arg types and didn't think that it was that great (mainly that 1.0 would become an integer). I'll stick out of this for now and let you finish it.